lib/internal/modules/esm/loader.js

```diff
@@ -96,6 +100,59 @@ const {
 
 let emittedSpecifierResolutionWarning = false;
 
+function nextHookFactory(chain, meta, validate) {
+  // First, prepare the current
+  const { hookName } = meta;
+  const {
+    fn: hook,
+    url: hookFilePath,
+  } = chain[meta.hookIndex];
+
+  const nextHookName = `next${
+    StringPrototypeToUpperCase(hookName[0]) +
+    StringPrototypeSlice(hookName, 1)
+  }`;
+
+  // When hookIndex is 0, it's reached the default, which does not call next()
+  // so feed it a noop that blows up if called, so the problem is obvious.
+  const generatedHookIndex = meta.hookIndex;
+  let nextNextHook;
+  if (meta.hookIndex > 0) {
+    // Now, prepare the next: decrement the pointer so the next call to the
+    // factory generates the next link in the chain.
+    meta.hookIndex--;
+
+    nextNextHook = nextHookFactory(chain, meta, validate);
+  } else {
+    // eslint-disable-next-line func-name-matching
+    nextNextHook = function chainAdvancedTooFar() {
+      throw new RangeError(
+        `ESM custom loader '${hookName}' advanced beyond the end of the chain; this is a bug in Node.js itself.`
+      );
```

Comment:
@aduh95 the linter doesn't like this (" Use an error exported by the internal/errors module"), but I don't see an existing error in internal/errors that seems appropriate, and this is sort of just a one-off, so it doesn't seem worth creating a new internal/errors for it. What should I do?

Reply:
I think you just need to create a new error in `internal/errors`, even though it’s only used by this.

Reply:
The definition for [RangeError](https://tc39.es/ecma262/multipage/fundamental-objects.html#sec-native-error-types-used-in-this-standard-rangeerror) is `Indicates a value that is not in the set or range of allowable values`. I don't think it's the correct error to throw here, probably `ERR_INTERNAL_ASSERTION` is more suited here.
To answer your question, you need to either create an error in `internal/errors`, or add a `// eslint-disable-next-line no-restricted-syntax` comment.

Reply:
The `this is a bug in Node.js itself` indicates to me we should use `ERR_INTERNAL_ASSERTION`, imo `RangeError` is for user-provided values, otherwise that's not a useful information to our users.